aten/src/ATen/native/cuda/linalg/BatchLinearAlgebra.cpp

```diff
@@ -2935,37 +2935,75 @@ static void lu_solve_kernel(const Tensor& LU, const Tensor& pivots, const Tensor
     return;
   }
 
-  // Summary: In most cases we use cublas / cusolver
-  // MAGMA is faster for large matrices, but it is buggy for trans != NoTranspose or for large batches.
-  // LU solve is fast in some cases when adjoint=True
+  // Heuristic
+  //if (n == k) {
```

Comment:
It is possible to compress this heuristic and/or express it as a function of `m * n * k`?